File: cpp/ISP/step1/Eagle.cpp
#include <string>
#include "Bird.cpp";
using namespace std;

class Eagle : public Bird {
public:
    Eagle(int initial_feather_count) {
        number_of_feathers_ = initial_feather_count;
    }

    void Fly() override {
        current_location_ = "in the air";
    }

    void Molt() override {
        number_of_feathers_ -= 1;
    }

private:
    string current_location_;
    int number_of_feathers_;
};

File: python/ISP/step1/eagle.py
from bird import Bird

class Eagle(Bird):
    def __init__(self, initial_feather_count) -> None:
        super().__init__()
        self.__current_location = ""
        self.__number_of_feathers = initial_feather_count

    def fly(self):
        self.__current_location = "in the air"

    def molt(self):
        self.__number_of_feathers -= -1
